### ISSUE TITLE
Replace RTM API with Event API for hubot slack adapter

### ISSUE THREAD

**Issue by yypptest** [slackapi/hubot-slack#630]
### Description

Describe your issue here.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [x] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/hubot-slack/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

This is the follow up issue for the issue #611. 

I noticed the response is there's no plan to update the hubot adapter to work with events-api instead of RTM. Could you pls share more background details for the decision? Is it because of the technical issues? or the direction? or the feature missing of the new APIs? Thanks!

#### Reproducible in:

hubot-slack version:

node version:

OS version(s):

#### Steps to reproduce:

1.
2.
3.

#### Expected result:

What you expected to happen

#### Actual result:

What actually happened

#### Attachments:

Logs, screenshots, screencast, sample project, funny gif, etc.

**Comment by srajiang** [comment]
@yypptest  - Thanks so much for writing in! 

As you already know, we don’t plan on updating the `hubot-slack` adaptor with `events-api` and recommend migrating your services to Bolt-JS if possible  ([here’s a migration guide](https://slack.dev/bolt-js/tutorial/hubot-migration) which may be helpful if you’re actively looking into this).

A while back the team got together and reviewed overall hubot-adaptor usage trends, the support for Hubot itself, which frameworks are the best ways to currently leverage the Slack platform features, and of course our limited resources as a small team of maintainers. We decided to go with the current recommendation considering all of those factors (and knowing that there are tradeoffs to every decision we make)!

So in short, all of the things you mentioned went into consideration. I hope that gives you a bit more of the context you’re looking for.

**Comment by yypptest** [comment]
thanks for your response.

**Comment by yypptest** [comment]
@srajiang Thanks for your response. 

So you mean from slack perspective, user is recommended to use Bolt-JS not the Hubot for slack integration, right?  And We are using Hubot embedded in the StackStorm, do you know if the Bolt-JS could be driven by StackStorm? Thannks!

**Comment by srajiang** [comment]
@yypptest - Hey there! Sorry for the delay. Yes, it is recommended to us Bolt-JS over the Hubot for slack integration. In terms of whether Bolt-JS could be driven by StackStorm, I assume that the [slack pack](https://github.com/StackStorm-Exchange/stackstorm-slack) doesn't match your use case. That would be a question to take to the StackStorm maintainers to see whether they will support.
